rpc/client.go

```diff
@@ -106,15 +107,35 @@ func NewClient(addr net.Addr, opts *retry.Options, context *Context) *Client {
 		Closed:       make(chan struct{}),
 		clock:        context.localClock,
 		remoteClocks: context.RemoteClocks,
-		cached:       !context.DisableCache,
 	}
-	if !context.DisableCache {
-		clients[c.Addr().String()] = c
+	clients[c.Addr().String()] = c
+
+	tlsConfig, tlsErr := context.GetClientTLSConfig()
+
+	connect := func() error {
+		if tlsErr != nil {
+			return tlsErr
+		}
+		return c.connect(tlsConfig)
+	}
+
+	var retryOpts retry.Options
+	if opts == nil {
+		retryOpts = clientRetryOptions
+	} else {
+		retryOpts = *opts
+	}
+	retryOpts.Tag = fmt.Sprintf("client %s connection", c.addr)
+	retryOpts.Stopper = context.Stopper
+
+	reconnect := func() error {
+		return retry.WithBackoff(retryOpts, func() (retry.Status, error) {
+			return retry.Continue, c.connect(tlsConfig)
```

Comment:
Yeah. For the most part, I think we want this (until we have rebalancing, at least), since the thing on the other end is known to have been alive at some point.